persistence-jdbc/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/persistence/jdbc/JdbcPersistenceComponents.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.scaladsl.persistence.jdbc
+
+import scala.concurrent.ExecutionContext
+
+import akka.actor.ActorSystem
+import com.lightbend.lagom.internal.persistence.jdbc.SlickProvider
+import com.lightbend.lagom.internal.scaladsl.persistence.ReadSideConfig
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.JdbcOffsetStore
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.JdbcReadSideImpl
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.OffsetTableConfiguration
+import play.api.Configuration
+import play.api.db.DBApi
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.JdbcPersistentEntityRegistry
+import com.lightbend.lagom.scaladsl.persistence.PersistentEntityRegistry
+import com.lightbend.lagom.internal.scaladsl.persistence.jdbc.JdbcSessionImpl
+import com.lightbend.lagom.internal.scaladsl.persistence.OffsetStore
+
+/**
+ * Persistence JDBC components (for compile-time injection).
+ */
+trait JdbcPersistenceComponents {
+
+  def system: ActorSystem
+  def executionContext: ExecutionContext
+  def configuration: Configuration
+  def dbApi: DBApi // FIXME where is this created?
```

Comment:
`play.api.db.DBComponents`

One question that I don't have the answer to is whether we should require the user to mix in `DBComponents` themselves, or whether this trait should extend `DBComponents`. Requiring them to mix it in offers more flexibility, but I'm not sure if that flexibility would ever really needed. It does make sense to require them to mix in `HikariCPComponents`, so that they can select a different connection pool if they need it.  But maybe not `DBComponents`.